### DISCUSSION TITLE
Rendering img differently if is child of another HTML component.

### DISCUSSION THREAD

**Discussion by gcponta** [remarkjs/remark#1030] [Q&A] [answered]
I am parsing the markdown to React like:

```
   const markdownContent = markdownRenderer()
    .use(shortcodes, { startBlock: '{{', endBlock: '}}', inlineMode: true })
    .use(reactRenderer, reactRendererOptions)
    .processSync(source).contents;

```
Where `reactRenderer` is the `remark-react` package.

And my `reactRendererOptions` is swapping some of the HTML to other custom components. In the case of `img` I am using a custom component that constantly updates the image URL in the img `src` to have the width of the parent container. At the moment it looks like this:

```
    img: props => {
      return <CustomImageComponent {...props} />;
    },
```

Therefore the image returned from the URL is always made to fit the container which can be quite large.

I have a case where I have icons that i want to pass as children of `p` tags and be smaller in size. But because of the custom component it is always expanded to the size of the container and therefore breaks the view. 

What can I pass inside the `reactRendererOptions` for `img` for it to know whether it is the child of the `p` tag. Something from props such as:

```
    img: props => {
      if( props.isChild) {
          return <img {...props} />
      }
      return <CustomImageComponent {...props} />;
    },

```
Or is there something extra I can pass in the markdown instead for it to be coming as a prop?

I also know that the `remark-react` package is out of support. If there a way I can do something like this with some of the new packages like `react-markdown` or `rehype-react`?

Thank you :)

**Comment by wooorm** [comment, marked as answer]
- `remark-react` is no longer recommended for use, use `rehype-react`.
- what is `markdownRenderer`?
- `contents` is no longer used, you are using old packages
- “constantly updates the image URL” this sounds like it will kick off a ton of requests to the server. I recommend using CSS instead to scale images.
- “What can I pass inside the reactRendererOptions for img for it to know whether it is the child of the p tag” React works the other way around: parents get children. Children don’t have access to parents. So you can’t do this.
- You seem to want to use different image components. In that case, MDX might be better. It has JSX syntax, so you can use a custom fancy image component for your resizing image, while all other images remain normal
